File: 2244-minimum-rounds-to-complete-all-tasks/2244-minimum-rounds-to-complete-all-tasks.cpp
class Solution {
public:
    int minimumRounds(vector<int>& tasks) {
        map<int,int>mp;
        for(int task:tasks){
            mp[task]++;
        }
        int num,freq=0,ans=0;
        for(auto i:mp){
            freq=i.second;
            if(freq==1){
                return -1;
            }
            else if(freq%3==0){
                ans+=freq/3;
            }
            else{
                ans+=freq/3+1;
            }
        }
        return ans;
    }
};

File: 2244-minimum-rounds-to-complete-all-tasks/2244-minimum-rounds-to-complete-all-tasks.java
class Solution {
    public int minimumRounds(int[] tasks) {
        Map<Integer, Integer> mp = new HashMap<>();
        for (int task: tasks) {
            mp.put(task, mp.getOrDefault(task, 0) + 1);
        }
        int freq, ans = 0;
        for (Map.Entry<Integer, Integer> entry: mp.entrySet()) {
            freq = entry.getValue();
            if (freq == 1) {
                return -1;
            }
            if (freq % 3 == 0) {
                ans += freq / 3;
            } else {
                ans += freq / 3 + 1;
            }
        }
        return ans;
    }
}